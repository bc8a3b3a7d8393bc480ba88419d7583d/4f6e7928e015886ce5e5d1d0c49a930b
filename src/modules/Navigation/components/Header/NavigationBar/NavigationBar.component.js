import React from 'react';
import PropTypes from 'prop-types';
import { View, ViewPropTypes } from 'react-native';

import styles from './NavigationBar.component.styles';

const NavigationBar = ({ children, style }) => (
  <View style={[styles.container, style]}>
    <View style={styles.statusBar} />
    <View style={styles.navigationBar}>{children}</View>
  </View>
);

NavigationBar.defaultProps = {
  style: {},
};

NavigationBar.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  style: ViewPropTypes.style,
};

export default NavigationBar;
