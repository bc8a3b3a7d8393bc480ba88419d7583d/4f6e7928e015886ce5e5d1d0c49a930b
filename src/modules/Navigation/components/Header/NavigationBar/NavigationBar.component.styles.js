import { StyleSheet } from 'react-native';
import { METRICS, COLORS } from '../../../../../themes';

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.background,
  },
  statusBar: {
    height: METRICS.statusBarHeight,
  },
  navigationBar: {
    height: METRICS.navigation.height,
  },
});

export default styles;
