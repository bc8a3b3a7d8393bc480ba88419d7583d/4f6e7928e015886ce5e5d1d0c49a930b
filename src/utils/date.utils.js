import moment from 'moment';

const supportedDateFormats = {
  ddddDDMMMYYYY: 'dddd, DD MMM YYYY',
};

const format = (date, dateFormat) => moment(date).format(dateFormat);

export default {
  format,
  supportedDateFormats,
};
