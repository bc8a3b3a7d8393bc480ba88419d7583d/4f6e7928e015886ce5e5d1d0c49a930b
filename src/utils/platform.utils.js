import { Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { equals } from 'ramda';

const isDEV = () => equals(process.env.NODE_ENV, 'development');
const isIOS = () => equals(Platform.OS, 'ios');
const isAndroid = () => equals(Platform.OS, 'android');
const iPhoneX = 'iPhone X';
const isIPhoneX = () => isIOS() && equals(DeviceInfo.getModel(), iPhoneX);

export { isDEV, isIOS, isAndroid, isIPhoneX, iPhoneX };
