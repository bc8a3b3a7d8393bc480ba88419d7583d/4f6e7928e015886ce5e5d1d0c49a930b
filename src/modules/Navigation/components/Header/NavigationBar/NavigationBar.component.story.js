import React from 'react';

// eslint-disable-next-line import/no-extraneous-dependencies
import { storiesOf } from '@storybook/react-native';

import { NavigationBar as Component } from './NavigationBar.component';

storiesOf('Navigation/Component - NavigationBar', module).add('default', () => (
  <Component
    navigation={{
      goBack: () => console.log('go back'),
    }}
  />
));
