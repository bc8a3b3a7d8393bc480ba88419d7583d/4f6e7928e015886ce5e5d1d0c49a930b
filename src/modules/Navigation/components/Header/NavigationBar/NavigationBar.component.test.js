import React from 'react';
import { View } from 'react-native';
import { testSnapshots } from '../../../../../utils/test.utils';

import NavigationBar from './NavigationBar.component';

const props = {
  navigation: {
    goBack: jest.fn(),
  },
  onPress: jest.fn(),
};

describe('NavigationBar', () => {
  testSnapshots(NavigationBar, [
    {
      props: {
        ...props,
        children: <View />,
      },
      description: 'basic render',
    },
    {
      props: {
        ...props,
        isRightButton: true,
        children: <View />,
      },
      description: 'render right button',
    },
  ]);
});
