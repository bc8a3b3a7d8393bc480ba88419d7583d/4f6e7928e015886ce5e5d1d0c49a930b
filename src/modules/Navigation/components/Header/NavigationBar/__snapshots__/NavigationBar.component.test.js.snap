// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar snapshots basic render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#2B9FDC",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "height": 20,
      }
    }
  />
  <View
    style={
      Object {
        "height": 44,
      }
    }
  >
    <View />
  </View>
</View>
`;

exports[`NavigationBar snapshots render right button 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#2B9FDC",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "height": 20,
      }
    }
  />
  <View
    style={
      Object {
        "height": 44,
      }
    }
  >
    <View />
  </View>
</View>
`;
